### PR TITLE
Stats: Update date picker logic to use URL params

### PR DESCRIPTION
**Pull request by a8ck3n**
Related to #83071 & #82221 & #83039.

## Proposed Changes

Updates the logic in the date picker. Shortcuts and input fields now update the chart via routing and URL params. Includes period computation from #83072. Thanks @grzegorz-cp!

### Test Initial State
1. Open the Live branch.
2. Navigate to **Stats → Traffic** page.
3. Confirm existing interval and navigation controls work.
4. Enable new date controls: `?flags=stats/date-control`
5. Confirm chart shows 7 days of data and button label shows corresponding dates.

### Test Shortcuts
1. Open the date picker and test various shortcuts.
2. Confirm the popover auto-close on selection.
3. Confirm the date labels update on selection
4. Confirm the chart updates to show the corresponding range.
5. Confirm selecting the "Last Year" shortcut sets the period to "month" in the URL.

### Test Input Fields
1. Open the date picker and test input fields.
2. Confirm the button label updates as the dates change. (needs updated logic)
3. Confirm entering a new range and hitting **Cancel** does not update the chart. (button label will show wrong dates)
4. Confirm entering a valid date range and hitting **Apply** updates things correctly.
5. Confirm that the period is updated to something reasonable if the date range is large.

Note: The date labels will get out of sync in step 3 as the labels are not driven by the URL params yet. They'll be correct again when updating the range and clicking **Apply**.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-date-control/stats-date-control-picker.tsx [hunk 1]
```diff
@@ -36,25 +29,23 @@ const DateControlPicker = ( {
...
-			addQueryPrefix: true,
-		} );
-		const period = 'day'; // TODO: make this dynamic
-		const url = `/stats/${ period }/${ slug }`;
-		const href = `${ url }${ nextDayQuery }`;
-
-		// expose the values externally
-		handleApply( inputStartDate, inputEndDate );
-
-		page( href );
```
Comment (kangzj) [inline client/my-sites/stats/stats-date-control/stats-date-control-picker.tsx:50]: Nice refactoring 👍

File: client/my-sites/stats/stats-date-control/stats-date-control-picker-date.tsx [hunk 2]
```diff
@@ -16,8 +16,8 @@ const DateControlPickerDate = ( {
 	return (
 		<div className="date-control-picker-date">
 			<div className="stats-date-control-picker-dates__inputs">
-				<TextControl value={ startDate } onChange={ onStartChange } />
-				<TextControl value={ endDate } onChange={ onEndChange } />
+				<TextControl type="date" value={ startDate } onChange={ onStartChange } />
```
Comment (kangzj) [inline client/my-sites/stats/stats-date-control/stats-date-control-picker-date.tsx]:
```suggestion
				<TextControl value={ startDate } onChange={ onStartChange } />
```

Good call to use date input, however `TextControl` does not support the type:

```
[13:22:45 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/11017322?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&showLog=11017322_7612_7612&logView=flowAware)
    client/my-sites/stats/stats-date-control/stats-date-control-picker-date.tsx(19,18): error TS2322: Type '"date"' is not assignable to type '"number" | "search" | "text" | "tel" | "url" | "email" | "password" | undefined'.

[13:22:45 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/11017322?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&showLog=11017322_7613_7613&logView=flowAware)
    client/my-sites/stats/stats-date-control/stats-date-control-picker-date.tsx(20,18): error TS2322: Type '"date"' is not assignable to type '"number" | "search" | "text" | "tel" | "url" | "email" | "password" | undefined'.
```

Let's leave it to another PR.

File: client/my-sites/stats/stats-date-control/stats-date-control-picker-date.tsx [hunk 3]
```diff
@@ -16,8 +16,8 @@ const DateControlPickerDate = ( {
...
+				<TextControl type="date" value={ endDate } onChange={ onEndChange } />
```
Comment (kangzj) [inline client/my-sites/stats/stats-date-control/stats-date-control-picker-date.tsx]:
```suggestion
				<TextControl value={ endDate } onChange={ onEndChange } />
```

**Review by kangzj — APPROVED** [review]
Looks solid 👍

File: client/my-sites/stats/stats-date-control/stats-date-control-picker-date.tsx (hunk 2 above)
Comment (a8ck3n) [reply client/my-sites/stats/stats-date-control/stats-date-control-picker-date.tsx]: Oops! I was meaning to remove this before pushing the branch. I added it to make entering dates easier/quicker while I was testing things out. Thanks for catching this @kangzj.
